apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/ResultStream.cs

```diff
@@ -135,7 +138,11 @@ private async Task<PartialResultSet> ComputeNextAsync(CancellationToken cancella
                     {
                         // Note: no cancellation token here; if we've been given a short cancellation token,
                         // it ought to apply to just the MoveNext call, not the original request.
-                        _grpcCall = _client.ExecuteStreamingSql(_request, _callSettings).GrpcCall;
+                        var sqlRequest = _request as ExecuteSqlRequest;
+                        var readRequest = _request as ReadRequest;
+                        _grpcCall = sqlRequest != null ?
+                            _client.ExecuteStreamingSql(sqlRequest, _callSettings).GrpcCall :
+                            _client.StreamingRead(readRequest, _callSettings).GrpcCall;
```

Comment:
```suggestion
                        _grpcCall = _request as ExecuteSqlRequest sqlRequest != null ?
                            _client.ExecuteStreamingSql(sqlRequest, _callSettings).GrpcCall :
                            _client.StreamingRead(_request as ReadRequest, _callSettings).GrpcCall;
```

Reply:
Maybe someting like this could be abstracted as well (either with the interface approach or the wrapping class approach), so that we could do:

```
_request.ExecuteStreaming(_client, _callSettings).GrcpCall;
```